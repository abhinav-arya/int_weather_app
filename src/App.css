home, html {
  width: 100%;
  height: 100%;
  font-family: 'Khand', sans-serif;
  font-size: 0.69em;
  padding: 0;
  margin: 0;
}

header {
  position: absolute;
  top: 0;
  z-index: 1;
  width: 100%;
  min-height: 4em;
  height: 7%;
  display: flex;
  flex-direction: row;
  flex-shrink: 1;
  justify-content: space-evenly;
  flex-wrap: wrap;
  align-items: center;
  background-color: #222;
  padding: 0.01em 0.5em 0.93em 2.65em;
  color: #fff;
}

header h3 {
  font-family: 'Ubuntu Condensed';
  margin: 0.9em 0.5em 0.93em 2.65em;
}

.search-btn form {
  position: relative;
  flex-shrink: 1;
  flex-wrap: nowrap;
}

.search-btn form input {
  width: 52rem;
  height: 3rem;
  font-family: 'Ubuntu Condensed', sans-serif;
  color: #666;
  border-radius: 1.3em;
  padding: 0.83em 0.5em 0.83em 2.65em;
  font-size: 1.3em;
  border: none;
}

.search-btn form img {
  position: absolute;
  top: 0.5em; 
  color: #666;
  width: 2rem;
  height: 2rem;
}

.search-btn form button {
  border-radius: 1.3em;
  position: relative;
  margin-left: 2rem;
  width: 12rem;
  height: 5rem;
  background: #380000;
  color: white;
  font-family: 'Ubuntu Condensed';
}

hr {
  display: block;
  height: 1.2px;
  width: 80%;
  background-color: #fff;
  margin-left: 0;
  border: none;
}

.weatherByTheHour, .renderedWeather {
  display: flex;
  flex-direction: row;
  color: #fff;
}

.hourlyWeatherItem {
  display: inline-block;
  font-size: 1.1em;
  padding: 3px 10px 3px 5px;
}

.hourlyWeatherItem img {
  width: 27px;
}

.rightPanel hr {
  margin-top: 1em;
  width: 90%;
}

.rightPanel .poweredBy {
  font-size: 2rem;
}

.renderedWeather {
  position: relative;
  height: 50%;
  min-height: 91vh;
  flex-shrink: 1;
  justify-content: space-evenly;
  align-items: center;
  background-blend-mode: darken;
  background: rgba(8, 8, 8, .68) url('https://source.unsplash.com/1920x1080/?rain') center center;
  background-size: cover;
  margin: 0 0 .2rem 0;
  padding: 10px;
}

.cityName {
  align-self: center;
  font-size: 6rem;
  margin: 4vw 3vw 1vw;
}

.todayWeather {
  position: relative;
  padding: 17% 0 18%;
  min-width: 100%;
  min-height: 100%;
}

.todayWeather, .todayWeather hr {
  margin: 10px;
  text-align: center;
}

.todayWeather div {
  font-size: 2.3rem;
}

.tNow, .todayWeather .tNow {
  font-size: 3.6rem;
  justify-self: center;
  margin: 0 0 1rem 0;
}

.leftPanel {
  flex-grow: 1;
}

.leftPanel hr {
  text-align: center;
  margin: 10px auto;
}

.leftPanel img {
  width: 4.4rem;
  margin-left: 19px;
}

.renderedWeather .cOrF {
  position: absolute;
  top: 15px;
  left: 15px;
  background-color: transparent;
  color: #fff;
  font-size: 1.3rem;
  padding: 5px;
  border: 2px solid #fff;
  border-radius: 5px;
}

.poweredBy {
  position: absolute;
  right: 12px;
  bottom: 12px;
  font-size: 1.3rem;
  font-family: 'Ubuntu Condensed';
}

.poweredBy a {
  text-decoration: none;
  color: inherit;
}

.forecast {
    position: absolute;
    display: table;
    text-transform: uppercase;
    margin-top: 15em;
    width: 90%;
}

.block {
    display: table-cell;
    padding: 1.5em 0 0 0;
    text-align: center;
}

.high {
    font-weight: 300;
    margin: 0.25em 0;
}

.secondary {opacity: 0.7;}

@media screen and (max-width:1200px) {
  .renderedWeather {
      height: auto;
      background-blend-mode: darken;
      background:  rgba(8, 8, 8, .68) url('https://source.unsplash.com/720x405/?rain') center center;
      background-size: cover;
      padding: 10px 20px 30px 50px;
  }

  .renderedWeather, .rightPanel, .dailyWeather {
      display: block;
  }

  .rightPanel, .dailyWeather, .forecastSummary {
      max-width: 96%;
  }

  .cityName {
      margin: 9vw 1vw 2vw;
      font-size: 5rem;
  }
}

@media screen and (max-width:900px) {

  .todayWeather div {
      font-size: 1.8rem;
  }

  .search-btn form button {
    position: relative;
    margin-top: 2rem;
  }

  .search-btn form input {
    width: 80%;
    height: 4rem;
  }
}

@media screen and (max-width:600px) {
  .renderedWeather {
      padding: 10px;
      background: linear-gradient(280deg, rgba(4, 0, 255, 1) 0%, rgba(30, 81, 125, 1) 100%);
  }

  .todayWeather div {
      font-size: 1.7rem;
  }

  .renderedWeather .cOrF {
      top: 15px;
      left: 72%;
      font-size: 0.9rem;
  }

  .icon img {
      width: 4rem;
      margin-bottom: 8px;
  }

  .poweredBy {
      margin-top: 10px;
      font-size: 1rem;
  }

  header {
    min-height: 12%;
    height: 12%;
    flex-direction: column;
  }

  .renderedWeather {
    background:  rgba(8, 8, 8, .68) url('https://source.unsplash.com/720x405/?rain') center center;
  }

  header h3 i {
    display: block;
    position: center;
    font-size: 0.7em;
    padding-top: 0.3em;
  }

  header h3 b{
    display: none
  }

  .search-btn form {
    display: block;
  }

  .search-btn form button {
    position: relative;
    margin-top: 2rem;
  }

  .search-btn form input {
    width: 80%;
    height: 4rem;
  }

  .forecast {
    margin-top: 0.2em;
    width: 95%;
  }

  .block {
      padding: 1em 0 0 0;
  }

  .high {
      font-weight: 200;
      margin: 0.15em 0;
  }
}
