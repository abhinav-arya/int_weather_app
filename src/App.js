import React from 'react';
import './App.css';
import MainPage from './components/MainPage'
// import { Navbar, Nav, NavDropdown, Form,  FormControl, Button, Table} from 'react-bootstrap'
import 'bootstrap/dist/css/bootstrap.min.css'

function App() {
  return (
    <div className="App">
      <header>
        <h3>The Weather App <b>-</b> <i>Powered by OpenWeatherMap</i></h3> 
      </header>
      <MainPage/>
    </div>
  );
}

export default App;
