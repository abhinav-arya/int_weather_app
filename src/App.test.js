import React from 'react';
import { shallow, configure } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import App from './App';

configure({adapter: new Adapter()})

test('renders Weather App Page', () => {
  const output = shallow(
      <App />
  )
  expect(shallowToJson(output)).toMatchSnapshot()
});
