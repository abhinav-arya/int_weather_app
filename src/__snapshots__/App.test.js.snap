// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Weather App Page 1`] = `
<div
  className="App"
>
  <header>
    <h3>
      The Weather App 
      <b>
        -
      </b>
       
      <i>
        Powered by OpenWeatherMap
      </i>
    </h3>
  </header>
  <MainPage />
</div>
`;
